README.md

```diff
@@ -2,14 +2,41 @@
 
 Command-line scripts to add data to a PDC service instance.
 
-## Available scripts
+## Separate scripts
 
-Each script should have a corresponding `npm` command. To list scripts, use `npm run -l`.
-
-## To run a script
+`generateBaseFieldsInserts` and `generateApplicationFormJson`
+each have a corresponding `npm` command.
 
 Run a script with a `--` between the name and its arguments:
 
 `npm run [script name] -- --[parameter name]=[arg]`
 
 To find information on one of the scripts, look in the script itself at `src/[script name].ts`.
+
+## Unified scripts
+
+Additional scripts are created as subcommands of a single, unified script.
+Run it with `npm start` to see the list of commands and required options.
+Pass arguments with a `--` between the name and its arguments: `npm start -- help`.
```

Comment:
I get three different results for `npm run start --help`, `npm start --help`, and `npm start -- help`, the last of which is the weirdest. I have a hard time recalling whether to run `npm run start` or `npm start` and what the difference is. So I don't know if a better command is available. But regardless I like the approach.

Reply:
> I get three different results for `npm run start --help`, `npm start --help`, and `npm start -- help`, the last of which is the weirdest. 

The first is giving you help for `npm run`, as it is ignoring `start` as a value for `<command>`. The second is giving you help for `npm start`, which is the preferred way to run a package in a dev environment. The third is passing the `help` argument to the script added in this PR, and can also be written `npm start -- --help` - both are auto-generated by yargs.

> I have a hard time recalling whether to run `npm run start` or `npm start` and what the difference is. So I don't know if a better command is available.

The `start`, `stop`, `restart`, and `test` commands are shortcuts to common scripts. `npm start` and `npm run start` are effectively the same, and either can be used here - but both require the `--` argument to pass subsequent arguments to the script itself instead of to npm.

> But regardless I like the approach.

Thanks! :blue_heart:

Reply:
`npm start` is a special npm command that invokes whatever the defined `start` script is according to `package.json`
`npm run {NAME_OF_SCRIPT}` will run the script defined in `package.json` with `NAME_OF_SCRIPT`

`npm {command} {-flag}` will pass the flag / parameter values to `npm` itself
`npm {command} -- {-flag}` will pass the flag / parameter values after `--` to the called script

(hopefully I didn't make any mistakes)

Reply:
Perhaps I made it sound like I personally was confused but my point is that npm makes it confusing. Of course I can look it up and dig around or try to memorize these distinctions. Thanks for taking the time to explain, though.

Reply:
I was able to run `npm start -- auth ...` with the appropriate args and get a token. Nice.

Reply:
[yargs says](https://github.com/yargs/yargs#description) it provides bash completion, but when I type `npm start -- auth --[tab]` it's only looking for files. Not that I need it, but I was trying to see what goodies came out of the box with yargs.

Reply:
[Completion](https://github.com/yargs/yargs/blob/e517318cea0087b813f5de414b3cdec7b70efe33/docs/api.md#completioncmd-description-fn) is an opt-in feature, but that's worth playing with - thanks for calling it out!

Reply:
Okay, I've spent some time playing with completion, and I think it's not worth doing, for a few reasons.

First, enabling bash completion involves evaluating some yargs-generated shell script in your current shell. This is something we'd want to keep up to date, which might be achievable with [direnv](https://direnv.net/) or the like, but is a bit risky - I don't love the idea of blindly executing external code upon entering a directory. Alternatively, each developer can periodically run `npm start -- completion >> ~/.bashrc` to pick up recently added commands, which is more secure and more likely to go out of date.

The other reason is that it'll take some doing to make the above work correctly. The way we're running this script is via `npm run-script`, and that adds some headers like `> data-scripts@1.0.0 start`. Unfortunately, the output of the `--get-yargs-completions` command - automatically added by `completion()` - is expected to be valid tab completions, and `>` is not a valid command. Additionally, the length of what should be `argv[0]` - `npm start --` makes it more complicated.

Finally, completion seems focused on commands, and not on options, which makes it less useful.

I think it could be made to work, but I also don't think it's worth the time. Which is unfortunate, because this is potentially a neat feature!

If you want to play with it further:

```diff
diff --git a/src/index.ts b/src/index.ts
index 61861b1..8320cf6 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -12,6 +12,7 @@ const main = async (argv: string[]) => yargs(hideBin(argv))
   .usage('Usage: npm start -- <command> [options]')
   .version(false)
   .strictCommands(true)
+  .completion()
   .env('DS')
   .config('config')
   .command(
```
```sh
npm start -- completion
```
```sh
npx ts-node src/index.ts --get-yargs-completions "candid"
```

Reply:
Thanks for looking into the completion and I agree if it's not safe or easy to enable, it's not worth it.

My unspoken thought was "what benefits do we get for the cost of the additional complexity associated with yargs?" I vaguely recall other benefits. Also if it is a tool that most Node.js developers use then it might make things easier for new developers.